### PR TITLE
test: Migrate 8 MachineHeaderForms test to RTL MAASENG-1730

### PR DESCRIPTION
**Pull request by ndv99**
## Done

Migrated the following files to RTL:
```
8.0K	./src/app/machines/components/MachineHeaderForms/AddChassis/AddChassisForm/AddChassisForm.test.tsx
8.0K	./src/app/machines/components/MachineHeaderForms/AddMachine/AddMachineFormFields/AddMachineFormFields.test.tsx
8.0K	./src/app/machines/components/MachineHeaderForms/MachineActionFormWrapper/CloneForm/CloneForm.test.tsx
8.0K	./src/app/machines/components/MachineHeaderForms/MachineActionFormWrapper/CloneForm/CloneFormFields/CloneFormFields.test.tsx
8.0K	./src/app/machines/components/MachineHeaderForms/MachineActionFormWrapper/CloneForm/CloneFormFields/CloneNetworkTable/CloneNetworkTable.test.tsx
8.0K	./src/app/machines/components/MachineHeaderForms/MachineActionFormWrapper/CommissionForm/CommissionForm.test.tsx
8.0K	./src/app/machines/components/MachineHeaderForms/MachineActionFormWrapper/ReleaseForm/ReleaseForm.test.tsx
8.0K	./src/app/machines/components/MachineHeaderForms/MachineActionFormWrapper/SetPoolForm/SetPoolForm.test.tsx
```

## Fixes

Fixes [MAASENG-1730](https://warthogs.atlassian.net/browse/MAASENG-1730)

[MAASENG-1730]: https://warthogs.atlassian.net/browse/MAASENG-1730?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/app/machines/components/MachineHeaderForms/AddMachine/AddMachineFormFields/AddMachineFormFields.test.tsx [hunk 1]
```diff
@@ -76,100 +77,104 @@ describe("AddMachineFormFields", () => {
...
+        }) as HTMLOptionElement
+      ).selected
+    ).toBe(false);
+    expect(
+      (
+        screen.getByRole("option", {
+          name: "No minimum kernel",
+        }) as HTMLOptionElement
+      ).selected
+    ).toBe(false);
```
Comment (petermakowski) [inline src/app/machines/components/MachineHeaderForms/AddMachine/AddMachineFormFields/AddMachineFormFields.test.tsx:105]:
```suggestion
    expect(
      screen.getByRole("option", {
        name: "bionic (ga-18.04)",
        selected: true,
      })
    ).toBeInTheDocument();
    expect(
      screen.getByRole("option", {
        name: "xenial (ga-16.04)",
        selected: false,
      })
    ).toBeInTheDocument();
    expect(
      screen.getByRole("option", {
        name: "No minimum kernel",
        selected: false,
      })
    ).toBeInTheDocument();
```

File: src/app/machines/components/MachineHeaderForms/MachineActionFormWrapper/CloneForm/CloneForm.test.tsx [hunk 2]
```diff
@@ -63,124 +66,187 @@ describe("CloneForm", () => {
...
+    ).not.toBeDisabled();
+
+    // Select config to clone - submit should be re-disabled.
+    await userEvent.click(
+      screen.getByRole("checkbox", { name: "Clone network configuration" })
+    );
+
+    expect(
+      screen.getByRole("checkbox", { name: "Clone network configuration" })
+    ).not.toBeDisabled();
```
Comment (petermakowski) [inline src/app/machines/components/MachineHeaderForms/MachineActionFormWrapper/CloneForm/CloneForm.test.tsx:106]: IMHO it's easier to read (totally optional, not fussed about it).

```suggestion
    ).toBeEnabled();
```

File: src/app/machines/components/MachineHeaderForms/MachineActionFormWrapper/CloneForm/CloneForm.test.tsx [hunk 3]
```diff
@@ -63,124 +66,187 @@ describe("CloneForm", () => {
...
+    expect(
+      screen.getByRole("button", { name: "Clone to machine" })
+    ).not.toBeDisabled();
```
Comment (petermakowski) [inline src/app/machines/components/MachineHeaderForms/MachineActionFormWrapper/CloneForm/CloneForm.test.tsx:109]:
```suggestion
    ).toBeEnabled();
```

File: src/app/machines/components/MachineHeaderForms/MachineActionFormWrapper/CloneForm/CloneFormFields/CloneFormFields.test.tsx [hunk 4]
```diff
@@ -117,35 +111,30 @@ describe("CloneFormFields", () => {
...
+        <CloneFormFields
+          selectedMachine={machine}
+          setSelectedMachine={jest.fn()}
+        />
+      </Formik>,
+      { state }
     );
-    const getTableClass = () =>
-      wrapper.find("MainTable.clone-table--network").prop("className");
+    let table = screen.getAllByRole("grid")[0];
```
Comment (petermakowski) [inline src/app/machines/components/MachineHeaderForms/MachineActionFormWrapper/CloneForm/CloneFormFields/CloneFormFields.test.tsx:128]: Provided you add an aria-label to `CloneNetworkTable`:

```suggestion
    let table = screen.getByRole("grid", { name: "Clone network" });
```

File: src/app/machines/components/MachineHeaderForms/MachineActionFormWrapper/CloneForm/CloneFormFields/CloneNetworkTable/CloneNetworkTable.test.tsx [hunk 5]
```diff
@@ -47,43 +42,53 @@ describe("CloneNetworkTable", () => {
...
-      </Provider>
-    );
-    expect(wrapper.find("MainTable").prop("rows")).toStrictEqual([]);
-    expect(wrapper.find("Placeholder").exists()).toBe(false);
+    renderWithMockStore(<CloneNetworkTable machine={null} selected={false} />, {
+      state,
+    });
+    expect(screen.getAllByRole("row")).toHaveLength(1);
+    // expect(wrapper.find("MainTable").prop("rows")).toStrictEqual([]);
+    // expect(wrapper.find("Placeholder").exists()).toBe(false);
```
Comment (petermakowski) [inline src/app/machines/components/MachineHeaderForms/MachineActionFormWrapper/CloneForm/CloneFormFields/CloneNetworkTable/CloneNetworkTable.test.tsx:50]:
```suggestion
```

Comment (petermakowski) [reply src/app/machines/components/MachineHeaderForms/MachineActionFormWrapper/CloneForm/CloneFormFields/CloneNetworkTable/CloneNetworkTable.test.tsx:50]: Missing placeholder assertion.

File: src/app/machines/components/MachineHeaderForms/MachineActionFormWrapper/CloneForm/CloneFormFields/CloneNetworkTable/CloneNetworkTable.test.tsx [hunk 6]
```diff
@@ -47,43 +42,53 @@ describe("CloneNetworkTable", () => {
...
+        machine={null}
+        selected={false}
+      />,
+      {
+        state,
+      }
     );
-    expect(wrapper.find("Placeholder").exists()).toBe(true);
+    const rows = screen.getAllByRole("row");
+    rows.shift(); // Remove the table header row
```
Comment (petermakowski) [inline src/app/machines/components/MachineHeaderForms/MachineActionFormWrapper/CloneForm/CloneFormFields/CloneNetworkTable/CloneNetworkTable.test.tsx:65]: Could you use an alternative way to query table body rows only that doesn't involve mutating the array (as I already suggested in another PR).

File: src/app/machines/components/MachineHeaderForms/MachineActionFormWrapper/CloneForm/CloneForm.test.tsx (hunk 2 above)
Comment (ndv99) [reply src/app/machines/components/MachineHeaderForms/MachineActionFormWrapper/CloneForm/CloneForm.test.tsx:106]: I genuinely didn't know about this until now... :facepalm:

**Review by petermakowski — APPROVED** [review]

**Pull request merged**